clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -247,12 +247,12 @@ public static void waitForCondition(TestCondition testCondition, String conditio
     public static void waitForCondition(TestCondition testCondition, long maxWaitMs, String conditionDetails) throws InterruptedException {
         long startTime = System.currentTimeMillis();
 
-
-        while (!testCondition.conditionMet() && ((System.currentTimeMillis() - startTime) < maxWaitMs)) {
+        boolean conditionMet;
+        while (!(conditionMet = testCondition.conditionMet()) && ((System.currentTimeMillis() - startTime) < maxWaitMs)) {
             Thread.sleep(Math.min(maxWaitMs, 100L));
         }
 
-        if (!testCondition.conditionMet()) {
+        if (!conditionMet) {
```

Comment:
Is this change really required? It seems like pretty much any test that requires it is going to be fragile as it implies that the condition may only be met for < 100ms. Normally I would expect anything using `waitForCondition` to expect that condition to switch to true and stay in that state until the test returns from `waitForCondition` and takes further action such that the condition could change.